packages/core/echo/echo-db/src/client/echo-client.ts

```diff
@@ -133,7 +134,14 @@ export class EchoClient extends Resource {
       throw err;
     }
 
-    const object = await db.loadObjectById(objectId, options);
-    return object;
+    if (localDataOnly) {
+      const objectDocId = db._coreDatabase._automergeDocLoader.getObjectDocumentId(objectId);
+      if (objectDocId?.split(':')[0] !== documentId) {
```

Comment:
not sure I follow the logic behind `objectDocId?.split(':')[0]`. The document ID should never contain any `:` characters

Reply:
right, that's some copy-pasted leftover